app/app.js

```diff
@@ -120,15 +125,143 @@ let Marvel = {
         showFirstLast: false
       })
     );
+  },
+  addRefinementList(){
+    this.search.addWidget(
+      instantsearch.widgets.currentRefinedValues({
+        container: '#current-refined-values',
+        clearAll: 'after'
+      })
+    );
+  },
+  renderHits(){
+    var renderedHits = {
+      render: function(options) {
+        processHeroProfile();
```

Comment:
I think there might be a better way. Here we are adding new bindings everytime a new set of results are displayed. Maybe we could just bind listeners on the parent container? Not blocking right now, but we might have to improve that for later.